truffle/src/main/ruby/core/complex.rb

```diff
@@ -53,9 +53,10 @@ class Complex < Numeric
   undef_method :i
 
   def self.convert(real, imag = undefined)
-    if real.equal?(nil) || imag.equal?(nil)
+    if nil.equal?(real) || nil.equal?(imag)
```

Comment:
Even though calling `literal.equal?(some_var)` is less nice, it's actually better for the semantics, otherwise you could invoke a method on a user object (more important for `==` than `equal?` which is rarely redefined). We could play some tricks in core, but I'd rather avoid too much magic unless needed.